server/src/OutputParser.ts

```diff
@@ -59,20 +63,33 @@ export class OutputParser {
   }
 
   reportProblems (): void {
+    if (_connection === null) {
+      // eslint-disable-next-line no-console
+      console.warn('The LSP Connection is not set. Dropping messages')
+      return
+    }
     logger.debug(`reportProblems: ${this._problems.length}`)
     this._problems.forEach((container: ProblemsContainer) => {
       logger.debug(`send Diagnostic ${container.toString()}`)
-      void this._connection.sendDiagnostics(container.getDignosticData())
+      void _connection?.sendDiagnostics(container.getDignosticData())
```

Comment:
I think we'll hate it when we'll try to use results from _connection and everything it produces might be undefined :/

Reply:
somehow TS couldn't correctly infer the type even though I already have a null check above. so I added this line before:
`const connection = _connection`. Now TS won't consider it as a possible null object.